instrumentation/log4j/log4j-2.13.2/library/src/main/java/io/opentelemetry/instrumentation/log4j/v2_13_2/OpenTelemetryLog4j.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.log4j.v2_13_2;
+
+import io.opentelemetry.sdk.logs.SdkLogEmitterProvider;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.concurrent.GuardedBy;
+
+public final class OpenTelemetryLog4j {
+
+  private static final Object lock = new Object();
+
+  @GuardedBy("lock")
+  private static SdkLogEmitterProvider sdkLogEmitterProvider;
+
+  @GuardedBy("lock")
+  private static final List<OpenTelemetryAppender> APPENDERS = new ArrayList<>();
+
+  public static void initialize(SdkLogEmitterProvider sdkLogEmitterProvider) {
+    List<OpenTelemetryAppender> instances;
+    synchronized (lock) {
+      if (OpenTelemetryLog4j.sdkLogEmitterProvider != null) {
+        throw new IllegalStateException("SdkLogEmitterProvider has already been set.");
+      }
```

Comment:
given trickiness around debugging init sequences, is it worth adding the stack trace of the original caller similar to GlobalOpenTelemetry?

https://github.com/open-telemetry/opentelemetry-java/blob/efbc87b4df38ea457c159b9360ae0db8a1791bb1/api/all/src/main/java/io/opentelemetry/api/GlobalOpenTelemetry.java#L82-L93

Reply:
I like that idea.